packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -336,6 +373,10 @@ export const queryBarReducer: Reducer<QueryBarState> = (
   if (isAction<ApplyQueryAction>(action, QueryBarActions.ApplyQuery)) {
     return {
       ...state,
+      fields: {
+        ...state.fields,
+        ...mapQueryToValidQueryFields(action.query),
```

Comment:
By updating the query value here it does mean that the user loses cursor position when their query is wrapped with the curly braces. This is something Sergey brought up in the ticket. Should we account for that?